src/app/shared/mock-data.service.ts

```diff
@@ -0,0 +1,241 @@
+
+  // persistence store, the MockDataService is a singleton when injected as a service.
+  private workItems: any[];
+  private workItemLinks: any[];
+  private workItemComments: any;
+
+  constructor() {
+    // create initial data store
+    this.workItems = this.workItemMockGenerator.createWorkItems();
+    this.workItemLinks = this.workItemMockGenerator.createWorkItemLinks();
+    this.workItemComments = this.workItemMockGenerator.createWorkItemComments();
+  }
+
+  // utility methods
+
+  private createId(): string {
+    var id = '';
+    var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+    for (var i = 0; i < 5; i++)
+      id += possible.charAt(Math.floor(Math.random() * possible.length));
+    console.log('Created new id ' + id);
+    return id;
+  }
+
+  private makeCopy(input: any): any {
+    return JSON.parse(JSON.stringify(input));
+  }
+
+  // data accessors
+
+  // work items and dependend entities (comments, ..)
+
+  public getWorkItems(): any {
+    return this.makeCopy(this.workItems);
+  }
+
+  public createWorkItemOrEntity(extraPath: string, entity: any): any {
+    if (extraPath && extraPath.indexOf('/') != -1) {
+      // request for subentities on a wi 
+      var parts = extraPath.split('/');
+      var wiId = parts[0];
+      var subselect = parts[1];
+      if (subselect === 'comments') {
+        console.log('Request create new comment for workitem ' + wiId);
+        var newId = this.createId();
+        var newComment = {
+              'attributes': {
+                'body': entity.data.attributes.body,
+                'created-at': '2000-01-01T09:00:00.000000Z'
+              },
+              'id': newId,
+              'links': {
+                'self': 'http://demo.api.almighty.io/api/comments/' + newId
+              },
+              'relationships': {
+                'created-by': {
+                  'data': {
+                    'id': 'user0',
+                    'type': 'identities'
+                  }
+                }
+              },
+              'type': 'comments'
+        };
+        if (this.workItemComments[wiId])
+          this.workItemComments[wiId].data.push(newComment);
+        else
+          this.workItemComments[wiId] = { data: [ newComment ] };
+        return { data: newComment };
+      }
+      return {};
+    }
+    var localWorkItem = this.makeCopy(entity.data);
+    localWorkItem.id = this.createId();
+    localWorkItem.links = { 
+          'self': 'http://mock.service/api/workitems/id' + localWorkItem.id
+        };
+    localWorkItem.relationships = { 
+          'assignees': { }, 
+          'baseType': { 
+            'data': { 
+              'id': 'system.userstory', 
+              'type': 'workitemtypes' 
+            } 
+          }, 
+          'comments': { 
+            'links': { 
+              'related': 'http://mock.service/api/workitems/id' + localWorkItem.id + '/comments',
```

Comment:
No, it is just to make is visual that it is a mock. The mock http will ignore the hostname.